pkg/nmi/server/server.go

```diff
@@ -216,6 +219,38 @@ func (s *Server) hostHandler(logger *log.Entry, w http.ResponseWriter, r *http.R
 	w.Write(response)
 }
 
+func (s *Server) isMIC(podNS, rsName string) bool {
+	micRegEx := regexp.MustCompile(`^mic-*`)
+	if strings.EqualFold(podNS, s.MICNamespace) && micRegEx.MatchString(rsName) {
+		return true
+	}
+	return false
+}
+
+func (s *Server) getMICToken(logger *log.Entry, rqClientID, rqResource string) ([]byte, int, error) {
+	var token *adal.Token
+	var err error
+	// ClientID is empty, so we are going to use System assigned MSI
+	if rqClientID == "" {
+		logger.Infof("Fetching MIC token for system assigned MSI")
+		token, err = auth.GetServicePrincipalTokenFromMSI(rqResource)
+	} else { // User assigned identity usage.
+		logger.Infof("Fetching MIC token for user assigned MSI for id: %s", rqResource)
+		token, err = auth.GetServicePrincipalTokenFromMSIWithUserAssignedID(rqClientID, rqResource)
+	}
+	if err != nil {
+		logger.Errorf("Failed to get service principal token. Error: %+v", err)
+		// TODO: return the right status code based on the error we got from adal.
+		return nil, http.StatusForbidden, err
+	}
+	response, err := json.Marshal(*token)
+	if err != nil {
+		logger.Errorf("Failed to unmarshal service principal token. Error: %+v", err)
+		return nil, http.StatusInternalServerError, err
+	}
+	return response, 0, nil
```

Comment:
I don't think 0 is a valid http status code. We should return maybe 200 OK, because request was successful.

Reply:
For success case, the return code is not used. We just write the response. In any case have changed it to http.StatusOK